cmd/thv-proxyrunner/app/run.go

```diff
@@ -0,0 +1,411 @@
+package app
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/stacklok/toolhive/pkg/kubernetes/config"
+	"github.com/stacklok/toolhive/pkg/kubernetes/container"
+	"github.com/stacklok/toolhive/pkg/kubernetes/logger"
+	"github.com/stacklok/toolhive/pkg/kubernetes/permissions"
+	"github.com/stacklok/toolhive/pkg/kubernetes/process"
+	"github.com/stacklok/toolhive/pkg/kubernetes/registry"
+	"github.com/stacklok/toolhive/pkg/kubernetes/runner"
+	"github.com/stacklok/toolhive/pkg/kubernetes/runner/retriever"
+	"github.com/stacklok/toolhive/pkg/kubernetes/transport"
+	"github.com/stacklok/toolhive/pkg/kubernetes/workloads"
+)
+
+var runCmd = &cobra.Command{
+	Use:   "run [flags] SERVER_OR_IMAGE_OR_PROTOCOL [-- ARGS...]",
+	Short: "Run an MCP server",
+	Long: `Run an MCP server with the specified name, image, or protocol scheme.
+
+ToolHive supports three ways to run an MCP server:
+
+1. From the registry:
+   $ thv run server-name [-- args...]
+   Looks up the server in the registry and uses its predefined settings
+   (transport, permissions, environment variables, etc.)
+
+2. From a container image:
+   $ thv run ghcr.io/example/mcp-server:latest [-- args...]
+   Runs the specified container image directly with the provided arguments
+
+3. Using a protocol scheme:
```

Comment:
I don't think the code calls the logic which handles this, nor do I think that it can be made to work in k8s. I think we should get rid of this.